models/repo_branch.go

```diff
@@ -58,6 +58,8 @@ func (repo *Repository) CheckoutNewBranch(oldBranch, newBranch string) error {
 type Branch struct {
```

Comment:
Could we move this to `modules/git`?

Reply:
@lunny maybe we can have it in a subsequent PR? This is already big. I can help the refactoring.

Reply:
Sorry for the delay. I will look into this soon. There was some performance regression related to listing branches and/or branch names and I want to address it too.

Reply:
I've moved the code around. Comments welcome.